docs/Adding-Software-to-Homebrew.md

```diff
@@ -8,46 +8,139 @@ Before you start, please check the open pull requests for [Homebrew/homebrew-cor
 
-1. You will now have to develop the boilerplate code from `brew create` into a full-fledged formula. Your main references will be the [Formula Cookbook](Formula-Cookbook.md), similar formulae in Homebrew, and the upstream documentation for your chosen software. Be sure to also take note of the Homebrew documentation for writing [Python](Python-for-Formula-Authors.md) and [Node](Node-for-Formula-Authors.md) formulae, if applicable.
+1. If you're starting from scratch, you can use the [`brew create` command](Manpage.md#create-options-url) to produce a basic version of your formula. This command accepts a number of options and you may be able to save yourself some work by using an appropriate template option like `--python` / `--go`.
 
-1. Make sure you write a good test as part of your formula. Refer to the [Add a test to the formula](Formula-Cookbook.md#add-a-test-to-the-formula) section of the Cookbook for help with this.
+1. You will now have to develop the boilerplate code from `brew create` into a complete formula. Your main references will be the [Formula Cookbook](Formula-Cookbook.md), similar existing formulae, and the official documentation of the software you're packaging. Be sure to also take note of the Homebrew documentation for writing [Python](Python-for-Formula-Authors.md) and [Node](Node-for-Formula-Authors.md) formulae, if applicable.
 
-1. Try installing your formula using `brew install --build-from-source <formula>`, where *\<formula>* is the name of your formula. If any errors occur, correct your formula and attempt to install it again. The formula installation should finish without errors by the end of this step.
+1. Make sure you write a good test as part of your formula. Refer to the [Add a test to the formula](Formula-Cookbook.md#add-a-test-to-the-formula) section of the Formula Cookbook for help with this.
 
 If you're stuck, ask for help on GitHub or the [Homebrew discussion forum](https://github.com/orgs/Homebrew/discussions). The maintainers are very happy to help but we also like to see that you've put effort into trying to find a solution first.
 
 ### Testing and auditing the formula
 
-1. Run `brew audit --strict --new-formula --online <formula>` with your formula. If any errors occur, correct your formula and run the audit again. The audit should finish without any errors by the end of this step.
+1. Test your formula installation using `brew install --formula --build-from-source $(brew --repository homebrew/core)/Formula/<prefix>/<formula>.rb`, where `<prefix>` is the first letter of your formula's name and `<formula>` is the name of your formula. Correct any errors that occur and repeat the installation until it completes without errors.
```

Comment:
No need to specify the full path here and it's actually undesirable to do so.

Reply:
I wasn't sure for Formula, but definitely for Casks I have been finding that since the JSON API changes, I have to explicitly point to the file path to be able to install my local changes. Which is why I found/opened the following issue while trying to find the canonical way of doing so:

- https://github.com/Homebrew/brew/issues/16325

If this 'just works' for Formula with the `--build-from-source` then happy to put it back as is; but not sure that will be relevant for Casks?

Reply:
Is it necessary for casks even with homebrew/cask tapped and `HOMEBREW_NO_INSTALL_FROM_API` set? Weird if so (and a bug we should fix).

Reply:
I possibly/probably didn't have `HOMEBREW_NO_INSTALL_FROM_API` set. And was avoiding it based on https://github.com/Homebrew/brew/issues/16325#issuecomment-1853254855

And because it wasn't mentioned anywhere in the `brew install -h` docs as called out in https://github.com/Homebrew/brew/issues/16325:

> If the `--local` isn't a good option, then perhaps drawing attention to `HOMEBREW_NO_INSTALL_FROM_API=1` (if that's the canonical way) in the text for `brew install -h` would be useful.

---

If that is the current canonical way that we have to do it, that's fine, and we can document that here and maybe also in `brew install -h` ; and then if/when `--local` lands, we could change it again.

Right now the situation basically feels like needing a PHD in Homebrew history and git deepdives to figure out what I need to use when/why though.

Reply:
@MikeMcQuaid Fixed in f934604734eb6a78230ea0b76ac2cbce4c6a3551

I'll leave this comment thread open for visibility for now; but once you check it and are happy, leave a comment and then feel free to mark it resolved.